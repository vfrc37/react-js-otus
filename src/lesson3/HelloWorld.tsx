import React from "react";

interface Prop {
  username: string;
}

export const HelloWorld: React.FC<Prop> = ({ username }) => (
  <h1>Hello World, {username}!</h1>
);
