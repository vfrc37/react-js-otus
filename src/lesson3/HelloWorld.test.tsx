import React from "react";
import { shallow } from "enzyme";

import { HelloWorld } from "./HelloWorld";

describe("HelloWorld render", function () {
  it("Render with prop username", function () {
    const wrapper = shallow(<HelloWorld username="UserName" />);
    expect(wrapper).toMatchSnapshot();
  });
});
