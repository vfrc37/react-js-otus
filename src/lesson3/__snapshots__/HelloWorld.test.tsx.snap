// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld render Render with prop username 1`] = `ShallowWrapper {}`;
